providers/gce/gce_instances.go

```diff
@@ -92,6 +93,58 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 	return refs
 }
 
+// cmpAddressFamily compares IPv6 and IPv4 address families so that IPv6 is smaller than IPv4.
+func cmpAddressFamily(a, b string) int {
+	// IPv6 addresses contain a colon („:”), IPv4 addresses do not.
+	if strings.Contains(a, ":") && !strings.Contains(b, ":") {
+		return -1
+	} else if !strings.Contains(a, ":") && strings.Contains(b, ":") {
+		return 1
+	}
+
+	return 0
```

Comment:
This effectively gives random order across entries from the same IP family. We should preserve the order within the family and then make sure that IP families are returned with the IP family precedence matching cluster stack type.

I would rather see it as splitting the original addresses slice into two slices and then merging the slices according to cluster stack type.

Reply:
I think the order would be random only if there are multiple addresses of the same family (IPv6/IPv4) and type (internal/external). Does that matter which internal IPv6 address is returned first, if there are many of them? Why does it matter? I can actually change to stable sorting, so the order would not change.

Reply:
I changed to stable sorting.

Reply:
There is a separate thread on sorting stability and/or merging slices, so I'll close this one.